### PR TITLE
enforce consistency in title + tests

### LINKED ISSUE #15: Easily enforce consistency in title
Make it easy to configure mergeable such that the title follows a certain format. For example, in previous projects we've required certain prefixes like `(feat)`, `(doc)`, `(fix)` in the title. This should be easily configured without complex regex like this https://github.com/jusx/mergeable/issues/12#issuecomment-386956281

Instead we should be able to do something like this:

```yml
title: 
   excludes: 'wip'
   includes: '^\(feat\)|^\(doc\)|^\(fix\)' 
```

Probably will still want to be backward compatible with:
```yml
title: 'wip'
```

### PR DESCRIPTION
**Pull request by shine2lay**
A second attempt at enforcing consistency in title by putting using a library to allow for re-usability and easily add more filters. 

@jusx I changed the `include` and `exclude` to `must-include` and `must-exclude` respectively to avoid confusion. 
Since we can also add many different filter, i think it make sense to show all the things that failed, I turned the `description` field into an array which holds all the errors. Backward compatibility is maintained so `description` can also be a string as well.

closes #15

### CODE REVIEW

File: __tests__/title.test.js [hunk 1]
```diff
@@ -2,6 +2,17 @@ const Helper = require('../__fixtures__/helper')
 const title = require('../lib/title')
 const Configuration = require('../lib/configuration')
 
+test('fail gracefully if invalid setting is provided', async () => {
+  // try to pass an array instead
+  let config = new Configuration(`
+    mergeable:
+      title: ['item 1', 'item 2' ]
```
Comment (jusx) [inline __tests__/title.test.js]: Interesting. Didn't know you can specify array in YML

File: lib/processor.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+module.exports = {
+  /**
+   *
+   * @param input the string run regex against
+   * @param filters string | object , if string we check if that regex is excluded from input
+   * @returns object { mergeable: bool, description: string | [string]}
+   */
+  processFilters: function (input, filters) {
```
Comment (jusx) [inline lib/processor.js]: We've been using es6 style for functions. Would be good to keep that standard throughout. 

`processFilters: () => {}`

Is it possible to configure standard to error on this I wonder?

File: lib/processor.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    const isMergeable = regex.test(input)
+    return { mergeable: isMergeable,
+      description: isMergeable ? null : `Title does not contain "${match}"`
+    }
+  },
+  mustExclude: function (input, match) {
+    const regex = new RegExp(match, 'i')
+    const isMergeable = !regex.test(input)
+
+    return { mergeable: isMergeable,
```
Comment (jusx) [inline lib/processor.js]: May want to add carriage return here for consistency throughout code base :-)

Comment (shine2lay) [reply lib/processor.js]: done

File: lib/processor.js (hunk 2 above)
Comment (shine2lay) [reply lib/processor.js]: done, changed it to `const processFilters = () => {}`

**Pull request merged**